### DISCUSSION TITLE
BusinessRules.CheckRules() throwing 'Object reference not set to an instance of an object'

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3365] [Questions] [answered]
We have a Csla 4.xx project. One of our parent objects (that has 'child'objects ) is throwing the above error in BusinessBase.DataPortal_Create()
I have the following stack trace

 ```cs
       BusinessRules.RunRules(IEnumerable`1 rules, Boolean cascade, RuleContextModes 
       executionContext) line 739
       BusinessRules.CheckRulesForProperty(IPropertyInfo property, Boolean cascade, RuleContextModes 
       executionContext) line 573
       BusinessRules.CheckRules(IPropertyInfo property, RuleContextModes executionContext) line 522
       BusinessRules.CheckRules() line 421
       BusinessBase.DataPortal_Create() line 1313
  ```

Kind Regards

**Comment by rockfordlhotka** [comment]
That doesn't provide the actual exceeption though.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  
I narrowed it down to 
  {"Property load or set failed for property EmployeeID (Object reference not set to an instance of an object.)"}

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  We have inheritance in the object graph where EmployeeID is a property of abstract base class.
We only have business properties and AddBusinessRules method is our abstract base class (always have been the same).
Do we need to change our abstract base class?

**Reply by Chicagoan2016** [reply]
In our concrete object the base.DataPortal_Create() is throwing the Object reference not set to an instance of a reference object.
The base in our concrete class referes to the abstract base class (that has only properties and AddBusinessRules).
It's not initializing all of a sudden. It was working I promise : )

**Reply by Chicagoan2016** [reply, marked as answer]
It was an error in our business rules using Priorities, A business rule with lower priority was called which I thought shouldn't be the case. I am going to start another thread.
